File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/info/AudioTrackInfoProvider.java
package com.sedmelluq.discord.lavaplayer.track.info;

/**
 * Provider for audio track info.
 */
public interface AudioTrackInfoProvider {
  /**
   * @return Track title, or <code>null</code> if this provider does not know it.
   */
  String getTitle();

  /**
   * @return Track author, or <code>null</code> if this provider does not know it.
   */
  String getAuthor();

  /**
   * @return Track length in milliseconds, or <code>null</code> if this provider does not know it.
   */
  Long getLength();

  /**
   * @return Track identifier, or <code>null</code> if this provider does not know it.
   */
  String getIdentifier();

  /**
   * @return Track URI, or <code>null</code> if this provider does not know it.
   */
  String getUri();
}

File: main/src/main/java/lavaplayer/track/info/AudioTrackInfoProvider.kt
package lavaplayer.track.info

/**
 * Provider for audio track info.
 */
interface AudioTrackInfoProvider {
    /**
     * @return Track title, or `null` if this provider does not know it.
     */
    val title: String?

    /**
     * @return Track author, or `null` if this provider does not know it.
     */
    val author: String?

    /**
     * @return Track length in milliseconds, or `null` if this provider does not know it.
     */
    val length: Long?

    /**
     * @return Track identifier, or `null` if this provider does not know it.
     */
    val identifier: String?

    /**
     * @return Track URI, or `null` if this provider does not know it.
     */
    val uri: String?

    /**
     * @return Track Artwork URL, or `null` if this provider does not know it.
     */
    val artworkUrl: String?
}
